client/signup/main.jsx

```diff
@@ -329,18 +330,45 @@ const Signup = React.createClass( {
 		// redirect the user to the next step
 		scrollPromise.then( () => {
 			if ( ! this.isEveryStepSubmitted() ) {
-				page( utils.getStepUrl( this.props.flowName, stepName, stepSectionName, this.props.locale ) );
+				if ( 'skip' === abtest( 'skipThemesSelectionModal' ) && 'themes' === stepName ) {
+					const flowSteps = flows.getFlow( this.props.flowName, stepName ).steps;
+					const currentStepIndex = indexOf( flowSteps, stepName );
+					const nextStepName = flowSteps[ currentStepIndex + 1 ];
+					const nextProgressItem = this.state.progress[ currentStepIndex + 1 ];
+					const nextStepSection = nextProgressItem && nextProgressItem.stepSectionName || '';
+
+					const designType = this.props.signupDependencies && this.props.signupDependencies.designType;
+
+					if ( designType ) {
+						const themeSlugWithRepo = ( ( type ) => {
+							switch ( type ) {
+								case 'blog': return 'pub/independent-publisher-2';
+								case 'grid': return 'pub/altofocus';
+								case 'page': return 'pub/dara';
+								default: return 'pub/twentyseventeen';
+							}
+						} ).call( this, designType );
+
+						SignupActions.submitSignupStep( { stepName: 'themes', wasSkipped: true }, [], { themeSlugWithRepo } );
+					}
+
+					page( utils.getStepUrl( this.props.flowName, nextStepName, nextStepSection, this.props.locale ) );
```

Comment:
if we make this `return page(...` we can get rid of the `else` clause below and leave the existing `page()` call at its original indentation level.